src/test/java/org/jabref/logic/bibtex/comparator/FieldComparatorTest.java

```diff
@@ -135,42 +135,134 @@ public void compareTypeFieldBiggerDescending() throws Exception {
         FieldComparator comparator = new FieldComparator(StandardField.TITLE);
-        BibEntry equal = new BibEntry();
-        equal.setField(StandardField.TITLE, "title");
+        BibEntry equal = new BibEntry()
+                .withField(StandardField.TITLE, "title");
 
         assertEquals(0, comparator.compare(equal, equal));
     }
 
     @Test
     public void compareStringFieldsEquality() throws Exception {
         FieldComparator comparator = new FieldComparator(StandardField.TITLE);
-        BibEntry equal = new BibEntry();
-        equal.setField(StandardField.TITLE, "title");
-        BibEntry equal2 = new BibEntry();
-        equal2.setField(StandardField.TITLE, "title");
+        BibEntry equal = new BibEntry()
+                .withField(StandardField.TITLE, "title");
+        BibEntry equal2 = new BibEntry()
+                .withField(StandardField.TITLE, "title");
 
         assertEquals(0, comparator.compare(equal, equal2));
     }
 
     @Test
     public void compareStringFieldsBiggerAscending() throws Exception {
         FieldComparator comparator = new FieldComparator(StandardField.TITLE);
-        BibEntry bigger = new BibEntry();
-        bigger.setField(StandardField.TITLE, "b");
-        BibEntry smaller = new BibEntry();
-        smaller.setField(StandardField.TITLE, "a");
+        BibEntry bigger = new BibEntry()
+                .withField(StandardField.TITLE, "b");
+        BibEntry smaller = new BibEntry()
+                .withField(StandardField.TITLE, "a");
 
         assertEquals(1, comparator.compare(bigger, smaller));
     }
 
     @Test
     public void compareStringFieldsBiggerDescending() throws Exception {
         FieldComparator comparator = new FieldComparator(new OrFields(StandardField.TITLE), true);
-        BibEntry bigger = new BibEntry();
-        bigger.setField(StandardField.TITLE, "a");
-        BibEntry smaller = new BibEntry();
-        smaller.setField(StandardField.TITLE, "b");
+        BibEntry bigger = new BibEntry()
+                .withField(StandardField.TITLE, "a");
+        BibEntry smaller = new BibEntry()
+                .withField(StandardField.TITLE, "b");
 
         assertEquals(1, comparator.compare(bigger, smaller));
     }
+
+    @Test
+    public void compareNumericFieldsIdentity() throws Exception {
+        FieldComparator comparator = new FieldComparator(StandardField.PMID);
+        BibEntry equal = new BibEntry()
+                .withField(StandardField.PMID, "123456");
+
+        assertEquals(0, comparator.compare(equal, equal));
+    }
+
+    @Test
+    public void compareNumericFieldsEquality() throws Exception {
+        FieldComparator comparator = new FieldComparator(StandardField.PMID);
+        BibEntry equal = new BibEntry()
+                .withField(StandardField.PMID, "123456");
+        BibEntry equal2 = new BibEntry()
+                .withField(StandardField.PMID, "123456");
+
+        assertEquals(0, comparator.compare(equal, equal2));
```

Comment:
Nearly all tests have the following form:

```suggestion
        FieldComparator comparator = new FieldComparator(StandardField.PMID);
        BibEntry ENTRY1 = new BibEntry()
                .withField(StandardField.PMID, ID1);
        BibEntry ENTRY2 = new BibEntry()
                .withField(StandardField.PMID, ID2);

        assertEquals(INT, comparator.compare(ENTRY1, ENTRY2));
```

Please convert these to a parameterized tests with INT, ID1, ID2 as input. (replace by `comparisonResult, id1, id2` in the implementation.

Hint:

```java
Stream.of(
  Arguments.of(0, "123456", "123456")
  ...
```

Full example at https://stackoverflow.com/a/61484155 (look at the second one 🌞 )